src/test/java/com/epam/tc/hw2/BaseTest.java

```diff
@@ -0,0 +1,41 @@
+package com.epam.tc.hw2;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import java.time.Duration;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+
+public class BaseTest {
+    public static WebDriver driver;
+
+    public static String getBaseURL() {
+        return baseURL;
+    }
+
+    private static final String baseURL = "https://jdi-testing.github.io/jdi-light/index.html";
+
+    @BeforeClass
+    public static void setupAll() {
+        WebDriverManager.chromedriver().setup();
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+        driver.get(baseURL);
+        new WebDriverWait(driver, Duration.ofSeconds(10))
+                .until(ExpectedConditions.presenceOfElementLocated(By.id("jdi-frame-site")));
+
+    }
+
+
+    @AfterClass
+    public static void tearDownDriver()  {
+        if (driver != null) {
+            driver.quit();
+        }
+    }
```

Comment:
Mostly every test class contains several test methods. Like 1 test class - 1 business feature. So, tests methods cover this feature.

Mostly you need a new browser for each test. Sometimes you need 1 browser for several tests in test class. For example for long business flow when each step is a test. Each test is based on previous test(step)

So, you should understand when use Before/After annotations

- Test - based on your test suite
- Method - for independent tests. Each test uses a new browser: `SuccesLogin, WrongPasswordLogin, WrongEmailLogin` 
- Class - for tests in class that have and use only 1 browser: `1)SuccesLogin, 2)AddToCarrito, 3)SuccesPayForStuff`

It's ok for now (studying), but always be sure when you need a new resource(browser)

Reply:
But please move `WebDriverManager.chromedriver().setup();` to BeforeSuit method. You need it only once.

Reply:
Fixed